crates/context/primitives/src/client/sync.rs

```diff
@@ -107,12 +107,12 @@ impl ContextClient {
                             .install_application_from_url(source, metadata, None)
                             .await?
                     }
-                    _ => {
-                        // fixme! we shouldn't assume both nodes run on the same machine
-                        self.node_client
-                            .install_application_from_path(source.path().into(), metadata)
-                            .await?
-                    }
+                    _ => self.node_client.install_application(
+                        &application.blob.bytecode,
+                        application.size,
+                        &source.into(),
+                        metadata,
+                    )?,
```

Comment:
as mentioned privately, if `from_url` fails, we should still fallback to normal install since "context config sync" shouldn't have a hard requirement of application blob availability